crates/swc_ecma_parser/src/lexer/state.rs

```diff
@@ -372,7 +353,7 @@ impl Iterator for Lexer<'_> {
             }
 
             self.state.update(start, token.kind());
-            self.state.prev_hi = self.last_pos();
+            self.state.prev_hi = self.input.cur_pos();
```

Comment:
We do not need maintain the cur_pos message in parser.lexer. How about we can get `span` from logos.lexer ?  It's a better way to reduce ecma.parser complexity.

Reply:
I agree that it can be a better design. You would only need to add `start_pos` to those.